ipc/provider/src/manager/evm/manager.rs

```diff
@@ -778,9 +783,123 @@ impl EthSubnetManager {
+            signatories,
+            cross_msgs,
+        } = bundle;
+
+        let address = contract_address_from_subnet(&checkpoint.subnet_id)?;
+        log::info!(
+            "submit bottom up checkpoint: {checkpoint:?} in evm subnet contract: {address:}"
+        );
+
+        let signatures = signatures
+            .into_iter()
+            .map(ethers::types::Bytes::from)
+            .collect::<Vec<_>>();
+        let signatories = signatories
+            .into_iter()
+            .map(|addr| payload_to_evm_address(addr.payload()))
+            .collect::<result::Result<Vec<_>, _>>()?;
+        let cross_msgs = cross_msgs
+            .into_iter()
+            .map(subnet_actor_manager_facet::CrossMsg::try_from)
+            .collect::<result::Result<Vec<_>, _>>()?;
+        let checkpoint = subnet_actor_manager_facet::BottomUpCheckpoint::try_from(checkpoint)?;
+
+        let signer = Arc::new(self.get_signer(submitter)?);
+        let contract =
+            subnet_actor_manager_facet::SubnetActorManagerFacet::new(address, signer.clone());
+        let call = contract.submit_checkpoint(checkpoint, cross_msgs, signatories, signatures);
+        call_with_premium_estimation(signer, call)
+            .await?
+            .send()
+            .await?;
+
+        Ok(())
+    }
+
+    async fn last_bottom_up_checkpoint_height(
+        &self,
+        subnet_id: &SubnetID,
+    ) -> anyhow::Result<ChainEpoch> {
+        let address = contract_address_from_subnet(subnet_id)?;
+        let contract = subnet_actor_getter_facet::SubnetActorGetterFacet::new(
+            address,
+            Arc::new(self.ipc_contract_info.provider.clone()),
+        );
+        let epoch = contract.last_bottom_up_checkpoint_height().call().await?;
+        Ok(epoch as ChainEpoch)
+    }
+
+    async fn checkpoint_period(&self, subnet_id: &SubnetID) -> anyhow::Result<ChainEpoch> {
+        let address = contract_address_from_subnet(subnet_id)?;
+        let contract = subnet_actor_getter_facet::SubnetActorGetterFacet::new(
+            address,
+            Arc::new(self.ipc_contract_info.provider.clone()),
+        );
+        let epoch = contract.bottom_up_check_period().call().await?;
+        Ok(epoch as ChainEpoch)
+    }
+
+    async fn checkpoint_bundle_at(
+        &self,
+        height: ChainEpoch,
+    ) -> anyhow::Result<BottomUpCheckpointBundle> {
+        let contract = gateway_getter_facet::GatewayGetterFacet::new(
+            self.ipc_contract_info.gateway_addr,
+            Arc::new(self.ipc_contract_info.provider.clone()),
+        );
+
+        let (checkpoint, _, signatories, signatures) =
+            contract.get_signature_bundle(height as u64).call().await?;
+        let cross_msgs = contract.bottom_up_messages(height as u64).call().await?;
+
+        let checkpoint = BottomUpCheckpoint::try_from(checkpoint)?;
+        let signatories = signatories
+            .into_iter()
+            .map(|s| ethers_address_to_fil_address(&s))
```

Comment:
Why do we need to perform this translation into fil address?

Reply:
Because in `ipc-agent` the convention is to use `fev_shared::Address`? We are removing this?